### PR TITLE
Added code for 0-1 BFS in cpp

### LINKED ISSUE #906: 0 1 BFS in cpp
Given a graph where every edge has weight as either 0 or 1. A source vertex is also given in the graph. Find the shortest path from source vertex to every other vertex.

### PR DESCRIPTION
**Pull request by susiejojo**
Fixes #906 
#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Sample Input / Output.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

#### Changes proposed in this pull request:

- Added 0-1_BFS.cpp

#### Languages Used:
- C++

#### Files Added:
- 0-1_BFS.cpp

### CODE REVIEW

File: 0-1BFS/0-1_BFS.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ void zeroOneBFS(int source,int n)
 				Q.push_back(i);
```
Comment (SakshiShreya) [inline 0-1BFS/0-1_BFS.cpp]: Replace tabs with 4 spaces.

File: 0-1BFS/0-1_BFS.cpp [hunk 2]
```diff
@@ -34,7 +34,7 @@ void zeroOneBFS(int source,int n)
 				Q.push_back(i); 
 			}
 		else
-			dist[i] = INT_MAX;
+			dist[i] = INT_MAX;//other vertices
```
Comment (SakshiShreya) [inline 0-1BFS/0-1_BFS.cpp]: - Add a space between code and comment
- Add a space after `\\`

File: 0-1BFS/0-1_BFS.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+using namespace std; 
+
+int dist[V];  //array to store the shortest distances
+//the graph stores vectors of int pairs such that graph[node].first = destination node & graph[node].second = edge weight
+vector < pair <int,int> > graph[V];
+deque <int> Q; // keeps track of edges during BFS
+
+//FUNCTIONS: 
+
+int distbfs(int i,int v)//returns the new distance to the destination node
```
Comment (SakshiShreya) [inline 0-1BFS/0-1_BFS.cpp]: Add space after `,`

File: 0-1BFS/0-1_BFS.cpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+	while (!Q.empty()) 
+	{ 
+		int v = Q.front(); 
+		Q.pop_front(); 
+
+		for (int i=0; i<graph[v].size(); i++) 
+		{ 
+			// checks for the shortest distance to the nodes
+			if (dist[graph[v][i].first] > distbfs(i,v))
```
Comment (SakshiShreya) [inline 0-1BFS/0-1_BFS.cpp]: Add space after `,`

File: 0-1BFS/0-1_BFS.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+{
+	return dist[v] + graph[v][i].second;
+}
+
+/* To print shortest distance from given source (here, 0) to 
+ every other node */
+void zeroOneBFS(int source,int n) 
+{ 
+	// Marking distances of all nodes from the source except itself as infinity  
+	int i=0;
```
Comment (SakshiShreya) [inline 0-1BFS/0-1_BFS.cpp]: Add space around `=`

File: 0-1BFS/0-1_BFS.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+//FUNCTIONS: 
+
+int distbfs(int i,int v)//returns the new distance to the destination node
+{
+	return dist[v] + graph[v][i].second;
+}
+
+/* To print shortest distance from given source (here, 0) to 
+ every other node */
+void zeroOneBFS(int source,int n)
```
Comment (SakshiShreya) [inline 0-1BFS/0-1_BFS.cpp]: Add space after `,`

File: Prefix_Sum/Prefix_Sum.c [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+//main function
+int main()
+{
+    int arr[] = { 1, 2, 3, 4, 5};
+    int n = sizeof(arr) / sizeof(arr[0]);
+    calculate_prefix_sum(arr, n);
+    for (int i = 0; i < n; i++)
+	printf ("%d ",prefixsum[i]);
```
Comment (SakshiShreya) [inline Prefix_Sum/Prefix_Sum.c]: Add space after `,`

File: Prefix_Sum/Prefix_Sum.c [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+/*function that calculates prefix sum by 
+forming the prefixsum array
+*/
+void calculate_prefix_sum (int arr[], int n)
+{
+    prefixsum[0] = arr[0]; //starts with the first element
+    for (int i = 1; i < n; i++)
+    {
+	/*calculates prefixsum by adding the new element
```
Comment (SakshiShreya) [inline Prefix_Sum/Prefix_Sum.c]: Replace tab with 4 spaces.

**Comment by jainaman224** [comment]
@susiejojo Please use [x] to fix checkboxes in PR description.

**Comment by susiejojo** [comment]
Commit : 44f9e2d

File: 0-1BFS/0-1_BFS.cpp [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+ every other node */
+void zeroOneBFS(int source,int n) 
+{ 
+    // Marking distances of all nodes from the source except itself as infinity  
+    int i = 0;
+    while (i < n) 
+    {
+	if (i == source)
+	{
+		dist[i] = 0; // distance of source from itself is 0
```
Comment (jainaman224) [inline 0-1BFS/0-1_BFS.cpp]: Please use 4 space indentation throughout the file.

Comment (susiejojo) [reply 0-1BFS/0-1_BFS.cpp]: Commit #7eb6633

**Comment by susiejojo** [comment]
Pls review the code as I am unable to upload codes for other issues I have been assigned.

**Comment by jainaman224** [comment]
@susiejojo You can add codes for other issues as well. No worries.
You can read about git branching if you are stuck.

File: 0-1BFS/0-1_BFS.cpp (hunk 9 above)
Comment (susiejojo) [reply 0-1BFS/0-1_BFS.cpp]: Can you pls help me about what changes to make?

**Comment by susiejojo** [comment]
Commit 689bd3a

**Comment by SakshiShreya** [comment]
Indentation is not fixed yet. Contact us on slack if you need any help with it.

**Comment by susiejojo** [comment]
As slack is for GSSoC participants, can we discuss the issues on gitter?

**Comment by jainaman224** [comment]
@susiejojo Are you still working on it?

**Comment by susiejojo** [comment]
Yes but I need some help identifying the errors in indentation..coz my editor supports 8 space indentation.

**Comment by jainaman224** [comment]
Let's have a conversation of gitter.

**Comment by MastersAbh** [comment]
I think this PR is good to merge now. @susiejojo just squash your commits.

**Pull request merged**